packages/@uppy/thumbnail-generator/src/index.js

```diff
@@ -161,6 +161,9 @@ export default class ThumbnailGenerator extends UIPlugin {
 
     return Promise.all([onload, orientationPromise])
       .then(([image, orientation]) => {
+        const originalImageDimensions = { width: image.width, height: image.height }
+        this.uppy.setFileMeta(file.id, { dimensions: originalImageDimensions })
```

Comment:
isn't this being stored somewhere else? is thumbnail-generator the right place to add this metadata? (it doesn't have anything to do with thumbnails I think?)

Reply:
What do you mean, stored somewhere else? Already, you mean? Not that I know of.

No, it’s not the right place, ideally we have a separate exif plugin that does this. But:
1. It’s available “for free” here, we already read the original image to memory, so I thought why not store useful data (some people have asked for it).
2. Alex asks the preview to behave differently, according to it’s size — if the image is smaller than container, like a tiny emoji, we have a white background for it, for instance. Ideally we need to know the image dimensions in CSS. So this would help here.

I also understand it doesn't feel “correct” to add this here, but since I was in this territory, decided to experiment and gather feedback.

Reply:
> What do you mean, stored somewhere else? Already, you mean? Not that I know of.

I just thought that width and height is very common metadata for an image so I thought we are already setting it somewhere else in the code.


> * It’s available “for free” here, we already read the original image to memory, so I thought why not store useful data (some people have asked for it).

Yea, but then if someone needs the dimensions, but they don't want the thumbnail plugin, then they won't get it. and if we later add this to core, then there will be two different width/height metadata, which would be confusing.

> 2\. Alex asks the preview to behave differently, according to it’s size — if the image is smaller than container, like a tiny emoji, we have a white background for it, for instance. Ideally we need to know the image dimensions in CSS. So this would help here.

Shouldn't it instead be stored inside plugin state then?

Reply:
> Shouldn't it instead be stored inside plugin state then, because it's a thumbnail plugin specific state?

👍